specification/cognitiveservices/OpenAI.Inference/models/completions/azure_chat_extensions.tsp

```diff
@@ -94,6 +99,51 @@ model AzureChatExtensionDataSourceResponseCitation {
   chunk_id?: string;
 }
 
+/** The retrieved document. */
+@added(ServiceApiVersions.v2024_05_01_Preview)
+model AzureChatExtensionRetrievedDocument {
+  ...AzureChatExtensionDataSourceResponseCitation;
+
+  /** The search queries used to retrieve the document. */
+  @encodedName("application/json", "search_queries")
+  searchQueries: string[];
+
+  /** The index of the data source. */
+  @encodedName("application/json", "data_source_index")
+  dataSourceIndex: int32;
+
+  /** The original search score of the retrieved document. */
+  @encodedName("application/json", "original_search_score")
+  originalSearchScore?: float64;
+
+  /** The rerank score of the retrieved document. */
+  @encodedName("application/json", "rerank_score")
+  rerankScore?: float64;
+
+  /**
+   * Represents the rationale for filtering the document. If the document does not undergo filtering,
+   * this field will remain unset.
+   */
+  @encodedName("application/json", "filter_reason")
+  filterReason?: AzureChatExtensionRetrieveDocumentFilterReason;
+}
+
+/** The reason for filtering the retrieved document. */
+@added(ServiceApiVersions.v2024_05_01_Preview)
+union AzureChatExtensionRetrieveDocumentFilterReason {
```

Comment:
Given that this is an output only, is it really an extensible enum? Can't the service guarantee the reason documents get filtered staying the same for a given API version?

Reply:
Removing `string` as a variant of this `union`